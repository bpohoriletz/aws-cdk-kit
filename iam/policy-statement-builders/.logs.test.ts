import LogsPolicyStatementBuilder from './logs';

describe('new LogsPolicyStatementBuilder', () => {
  let instance: LogsPolicyStatementBuilder;

  beforeEach(() => {
    instance = new LogsPolicyStatementBuilder();
  });

  test('has correct defaults set', () => {
    expect(instance).toMatchSnapshot();
  });

  test('.setUpload allows upload logs', () => {
    instance.setUpload();

    expect(instance).toMatchSnapshot();
  });

  test('.addResource adds resource', () => {
    instance.addResource('*');

    expect(instance).toMatchSnapshot();
  });
});
