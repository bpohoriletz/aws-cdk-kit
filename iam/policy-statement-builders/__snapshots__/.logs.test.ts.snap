// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new LogsPolicyStatementBuilder .addResource adds resource 1`] = `
LogsPolicyStatementBuilder {
  "policyStatementProps": PolicyStatementProduct {
    "actions": [],
    "resources": [
      "*",
    ],
  },
}
`;

exports[`new LogsPolicyStatementBuilder .setUpload allows upload logs 1`] = `
LogsPolicyStatementBuilder {
  "policyStatementProps": PolicyStatementProduct {
    "actions": [
      "logs:CreateLogStream",
      "logs:PutLogEvents",
    ],
    "resources": [],
  },
}
`;

exports[`new LogsPolicyStatementBuilder has correct defaults set 1`] = `
LogsPolicyStatementBuilder {
  "policyStatementProps": PolicyStatementProduct {
    "actions": [],
    "resources": [],
  },
}
`;
